### PR TITLE
Optimized validation

### PR DESCRIPTION
**Pull request by tlbdk**

### CODE REVIEW

**Review by mex — COMMENTED** [review]
No test case for the optimized version?

**Comment by tlbdk** [comment]
Right now there are no test cases for the non-optimized version :)

**Comment by mex** [comment]
What is https://github.com/connectedcars/node-object-validator/blob/master/index.test.js then?

**Comment by tlbdk** [comment]
class ObjectValidator {
  constructor(schema) {	  constructor(schema, **optimize = true**) {

**Comment by mex** [comment]
Ahh, so you just hijacked my tests?

**Review by mex — APPROVED** [review]

**Comment by tlbdk** [comment]
We need to do code coverage of the optimized generated code to ensure we catch all cases

File: .gitignore [hunk 1]
```diff
@@ -62,3 +62,5 @@ typings/
 
 # vscode configuration
 .vscode
+
+optimized.js
```
Comment (mex) [inline .gitignore:66]: Wouldn't it be nicer to generate this file in some more generic cache/build/dist folder and ignore that instead?

**Review by mex — APPROVED** [review]

**Pull request merged**